### PR TITLE
Implement new amendment majority semantics :

### PR DESCRIPTION
**Pull request by JoelKatz**
This implements the tracking of when an amendment achieved a majority in the ledger, ensuring that there's always network-wide agreement on which amendments have achieved a majority and how long they've held it.

* New fields
* Change transactor changes
* AmendmentTable API and implementation changes
* Update amendment enabled status on validated ledgers
* Reinstate support for ledger sequence in fee transactions

### CODE REVIEW

**Comment by seelabs** [comment]
Nonunity build is broken. You can cherry pick this commit to fix: https://github.com/seelabs/rippled/commit/d3dee05937fb02fc1f741a96235af9de4ddb5f17

File: src/ripple/app/misc/AmendmentTableImpl.cpp [hunk 1]
```diff
@@ -497,40 +385,39 @@ AmendmentTableImpl<AppApiFacade>::setSupported (const std::vector<uint256>& amen
...
 {
-    auto lAmendments = getDesired();
+    auto lAmendments = getDesired (enabledAmendments);
 
     if (lAmendments.empty())
-        return;
+        return {};
 
-    STVector256 amendments (sfAmendments);
+    std::vector <uint256> amendments;
```
Comment (seelabs) [inline src/ripple/app/misc/AmendmentTableImpl.cpp]: `amendments(lAmendments.begin(), lAmendments.end())` works too.

**Comment by rec** [comment]
I booted travis.

This seems to depend on work I have in a branch, which depends on work vinnie has in a branch. Is there a way to get past that?

We're reviewing Vinnie's branch right now...

File: src/ripple/app/misc/tests/AmendmentTable.test.cpp [hunk 2]
```diff
@@ -375,8 +377,401 @@ class AmendmentTable_test final : public beast::unit_test::suite
...
+        enabledAmendments_t enabled;
+        majorityAmendments_t majority;
+
+        for (int i = 0; i <= 17; ++i)
+        {
+            std::vector <std::pair <uint256, int>> votes;
+            std::vector <uint256> ourVotes;
+
+            if ((i > 0) && (i < 17))
+                votes.emplace_back (testAmendment, i * 16);
```
Comment (seelabs) [inline src/ripple/app/misc/tests/AmendmentTable.test.cpp]: @JoelKatz I don't understand the use of votes here. It looks like votes is per validator, but I think it should be per amendment. Looking at `AmendmentTable.test.cpp` line 423, doesn't this cause a validator to try to vote for the same amendment multiple times? Am I reading the code correctly?

File: src/ripple/ledger/impl/View.cpp [hunk 3]
```diff
@@ -387,6 +388,37 @@ cdirNext (BasicView const& view,
...
 
+enabledAmendments_t
+getEnabledAmendments (BasicView const& view)
+{
+    enabledAmendments_t amendments;
+    auto const sleAmendments = view.read(keylet::amendments());
+
+    if (sleAmendments)
+    {
+        for (auto const &a : sleAmendments->getFieldV256 (sfAmendments))
```
Comment (seelabs) [inline src/ripple/ledger/impl/View.cpp]: Nit: Formatting `auto const& a`

File: src/ripple/app/tx/impl/Change.cpp [hunk 4]
```diff
@@ -124,14 +125,66 @@ Change::applyAmendment()
...
+        }
+    }
+
+    if (! found && lostMajority)
+        return tefALREADY;
+
+    if (gotMajority)
+    {
+        // This amendment now has a majority
+        newMajorities.push_back (STObject (sfMajority));
```
Comment (seelabs) [inline src/ripple/app/tx/impl/Change.cpp]: Informational: If you use `emplace_back` you don't need the `STObject`, you can just say `newMajorities.emplace_back(sfMajority)`. Fine as is tho.

**Comment by seelabs** [comment]
Even though I only found nits, I did go thought this code for understanding and it looks correct. :+1:

File: src/ripple/ledger/impl/View.cpp (hunk 3 above)
Comment (vinniefalco) [inline src/ripple/ledger/impl/View.cpp]: The word "amendment" appears NINE times in this diff! Suggested:
`auto const sle = view.read(keylet::amendments());`

**Comment by vinniefalco** [comment]
Squashed and rebased

**Pull request closed without merging**